File: katalyze/src/main/java/messageHandlers/SingleMessageHandler.java
package messageHandlers;

import model.Contest;
import io.SimpleMessage;

public abstract class SingleMessageHandler implements MessageHandler {
	String tag;
	Contest contest;
	
	public SingleMessageHandler(String tag) {
		this.tag = tag;
	}
	
	/* (non-Javadoc)
	 * @see messageHandlers.MessageHandler#connectTo(model.Contest)
	 */
	@Override
	public void connectTo(Contest contest) {
		this.contest = contest;
	}
	
	/* (non-Javadoc)
	 * @see messageHandlers.MessageHandler#supports(io.SimpleMessage)
	 */
	@Override
	public Boolean supports(SimpleMessage message) {
		return tag.equals(message.getName());
	}
	
	/* (non-Javadoc)
	 * @see messageHandlers.MessageHandler#process(io.SimpleMessage)
	 */
	@Override
	public abstract void process(SimpleMessage message);
	
	public void error(String errorMessage) {
		System.out.println(errorMessage);
		
	}
	
	public void close() {
		// do nothing
	}

}

File: katalyze/src/main/kotlin/messageHandlers/SingleMessageHandler.kt
package messageHandlers

import legacyfeed.SimpleMessage
import model.Contest

abstract class SingleMessageHandler(var tag: String) : MessageHandler {
    var contest: Contest? = null

    /* (non-Javadoc)
	 * @see messageHandlers.MessageHandler#connectTo(model.Contest)
	 */
    override fun connectTo(contest: Contest) {
        this.contest = contest
    }

    /* (non-Javadoc)
	 * @see messageHandlers.MessageHandler#supports(legacyfeed.SimpleMessage)
	 */
    override fun supports(message: SimpleMessage): Boolean {
        return tag == message.name
    }

    /* (non-Javadoc)
	 * @see messageHandlers.MessageHandler#process(legacyfeed.SimpleMessage)
	 */
    abstract override fun process(message: SimpleMessage)

    fun error(errorMessage: String?) {
        println(errorMessage)
    }

    override fun close() {
        // do nothing
    }
}